src/core-sdk-tasks/RemoveDuplicateFiles.cs

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.IO.Enumeration;
+using System.IO.MemoryMappedFiles;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Replaces files that have the same content with hard links.
+    /// </summary>
+    public sealed class RemoveDuplicateFiles  : Task
```

Comment:
`RemoveDuplicateFiles` probably isn't the best name.  It's probably OK to re-use `ReplaceDuplicateFilesWithHardLinks`, which is what the target is named and is clearer, as the name for the task.